test/unit/mdc-snackbar/foundation.test.js

```diff
@@ -74,6 +77,30 @@ test('#destroy calls adapter.deregisterActionClickHandler() with a registerActio
   td.verify(mockAdapter.deregisterActionClickHandler(changeHandler));
 });
 
+test('#destroy calls adapter.deregisterVisibilityChangeHandler() with a function', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+
+  foundation.destroy();
+  td.verify(mockAdapter.deregisterVisibilityChangeHandler(isA(Function)));
```

Comment:
Deregistering the event handlers is not the concern of `destroy()` though. It is the concern of `adapter.deregisterVisibilityChangeHandler()`. Therefore, we should test that destroy calls that, and then in the test for `deregisterVisibilityChangeHandler()`, test that the event handlers are removed properly.